### PR TITLE
fix: AllowanceModule start-delay semantics and Safe module diagnostics

### PR DESCRIPTION
**Pull request by sherifahmed990**
Fixes for the Safe AllowanceModule and SocialRecoveryModule helpers.

  - **Delayed one-time allowances never worked**: `createOneTimeAllowanceMetaTransaction` passed `startAfterInMinutes` as the contract's `resetBaseMin` with `resetTimeMin = 0`, which reverts on-chain for any nonzero value (`setAllowance` explicitly requires `resetTimeMin > 0` whenever `resetBaseMin` is nonzero). The parameter is removed.
  - **Recurring allowance "delay" was misleading and unsafe**: the last parameter of `createRecurringAllowanceMetaTransaction` was documented as a relative activation delay, but  on-chain `resetBaseMin` is an absolute epoch-minutes baseline that must be in the past and only aligns period boundaries — a relative value made funds spendable immediately. Renamed to `inThePastPeriodStartBaseTimeStamp`, now taking a unix timestamp **in seconds** (converted to the contract's epoch-minutes `resetBaseMin` internally, flooring sub-minute precision). The docs state explicitly that the allowance is active immediately — the value only anchors the period accounting, with a worked example. A deterministic uint32 range check on the converted minutes catches millisecond-timestamp mistakes; the contract's own `require(resetBaseMin <= currentMin)` enforces the past bound at execution time, so a meta-transaction can be prepared before its anchor passes.
  - **Diagnostics**: `SocialRecoveryModule.getGuardians`/`.nonce` reported empty-result errors under the name "threshold"; `getRecoveryRequestEip712Data` only accepted a URL string  while every sibling accepts `string | Transport | JsonRpcNode`; `AllowanceModule.getDelegates` with `maxNumberOfResults > 255` threw an opaque ABI out-of-bounds error (the module's  page size is a uint8) and now fails with a clear `RangeError` pointing at pagination.

  Note: removing/renaming the start-delay parameters is a breaking change to those two helper signatures.

  Tests added in `test/safe/allowanceModule.test.js`; full offline suite passes.

## Summary by CodeRabbit

- **New Features**
  - Updated allowance meta-transaction creation: one-time allowances removed start-delay input and always use immediate reset values; recurring allowances now take an absolute baseline timestamp in seconds (defaulting to immediate) instead of a relative delay.

- **Bug Fixes**
  - Improved Social Recovery JSON-RPC helpers to use correct operation names for empty-result/nonce handling.
  - Added stricter validation for recurring baseline inputs and delegate pagination limits, throwing `RangeError` for out-of-range values.

- **Tests**
  - Added offline calldata decoding tests for allowance `setAllowance`, including baseline conversion edge cases and invalid-input rejection.

### CODE REVIEW

File: src/account/Safe/modules/AllowanceModule.ts [hunk 1]
```diff
@@ -41,51 +41,61 @@ export class AllowanceModule extends SafeModule {
...
 	 */
 	public createRecurringAllowanceMetaTransaction(
 		delegate: string,
 		token: string,
 		allowanceAmount: bigint,
 		recurringAllowanceValidityPeriodInMinutes: bigint,
-		startAfterInMinutes: bigint,
+		periodStartBaseInMinutes: bigint = 0n,
 	): MetaTransaction {
+		const nowInMinutes = BigInt(Math.floor(Date.now() / 60_000));
```
Comment (Sednaoui) [inline src/account/Safe/modules/AllowanceModule.ts]: The contract validates `resetBaseMin` against `block.timestamp` when the Safe transaction executes, while this guard uses the caller’s local clock when calldata is created. This can reject a valid queued transaction whose baseline will be in the past by execution time, or accept one that reverts if the local clock is ahead of the chain. Since this is a synchronous calldata builder, could we remove the `Date.now()` check and document that the baseline must be no later than the chain timestamp at execution? If immediate executability needs validation, that could be a separate async preflight using the latest block timestamp.

File: src/account/Safe/modules/AllowanceModule.ts [hunk 2]
```diff
@@ -41,51 +41,61 @@ export class AllowanceModule extends SafeModule {
...
+	 * and is spendable immediately once the Safe executes the transaction.
 	 * @param delegate - Address of the delegate to grant the allowance to.
 	 * @param token - ERC-20 token contract address (use zero address for native token).
 	 * @param allowanceAmount - Maximum amount per period, in the token's smallest unit.
 	 * @param recurringAllowanceValidityPeriodInMinutes - Duration of each allowance period in minutes.
-	 * @param startAfterInMinutes - Delay in minutes before the first allowance period begins.
+	 * @param periodStartBaseInMinutes - Optional absolute baseline that period
+	 * boundaries are aligned to, in minutes since the unix epoch (not a relative
+	 * delay — the module has no delayed-activation concept and requires this
+	 * baseline to be in the past). Defaults to 0n, aligning periods to the epoch.
```
Comment (Sednaoui) [inline src/account/Safe/modules/AllowanceModule.ts]: `0n` does not align periods to the epoch. The contract initializes a new allowance to the execution block’s current minute, while an existing allowance keeps its previous `lastResetMin`. Could we update this documentation accordingly?

File: src/account/Safe/modules/AllowanceModule.ts (hunk 1 above)
Comment (sherifahmed990) [reply src/account/Safe/modules/AllowanceModule.ts]: done

File: src/account/Safe/modules/AllowanceModule.ts (hunk 2 above)
Comment (sherifahmed990) [reply src/account/Safe/modules/AllowanceModule.ts]: done
